src/Core/AppKernel.php

```diff
@@ -0,0 +1,416 @@
+<?php
+
+namespace SilverStripe\Core;
+
+use Monolog\Handler\StreamHandler;
+use Monolog\Logger;
+use Psr\Log\LoggerInterface;
+use SilverStripe\Config\Collections\CachedConfigCollection;
+use SilverStripe\Control\Director;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Control\HTTPResponse_Exception;
+use SilverStripe\Core\Cache\ManifestCacheFactory;
+use SilverStripe\Core\Config\ConfigLoader;
+use SilverStripe\Core\Config\CoreConfigFactory;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Core\Injector\InjectorLoader;
+use SilverStripe\Core\Injector\SilverStripeServiceConfigurationLocator;
+use SilverStripe\Core\Manifest\ClassLoader;
+use SilverStripe\Core\Manifest\ClassManifest;
+use SilverStripe\Core\Manifest\ModuleLoader;
+use SilverStripe\Core\Manifest\ModuleManifest;
+use SilverStripe\Dev\DebugView;
+use SilverStripe\Dev\Install\DatabaseAdapterRegistry;
+use SilverStripe\Logging\ErrorHandler;
+use SilverStripe\ORM\DB;
+use SilverStripe\View\ThemeManifest;
+use SilverStripe\View\ThemeResourceLoader;
+
+class AppKernel extends CoreKernel
```

Comment:
If we call the core class `AppKernel`, what would a subclass of this be called in a default installer project? In Symfony, `AppKernel extends Kernel`, and `AppKernel.php` lives in project code. I think the `App` naming prefix should stay reserved for project code. How do you see `CoreKernel` being used other than extended through `AppKernel`? I'm wondering if all code currently in `AppKernel` should simply move into `CoreKernel`.